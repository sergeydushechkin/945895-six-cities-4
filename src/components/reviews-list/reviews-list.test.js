import React from "react";
import renderer from "react-test-renderer";
import ReviewsList from "./reviews-list.jsx";

import users from "../../mocks/tests_users.js";

const reviews = [
  {
    id: 1,
    userId: 2,
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    rating: 4.3,
    date: `2019-04-24`
  },
  {
    id: 2,
    userId: 3,
    text: `Very good`,
    rating: 3.2,
    date: `2019-04-25`
  }
];

it(`Render ReviewList`, () => {
  const tree = renderer
    .create(
        <ReviewsList
          reviews={reviews}
          users={users}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
