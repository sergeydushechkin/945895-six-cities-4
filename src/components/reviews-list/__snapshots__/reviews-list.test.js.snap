// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ReviewList 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Max
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
      </p>
      <time
        className="reviews__time"
        dateTime="2019-04-24"
      >
        2019-04-24
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Ivan
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Very good
      </p>
      <time
        className="reviews__time"
        dateTime="2019-04-25"
      >
        2019-04-25
      </time>
    </div>
  </li>
</ul>
`;
